pywbem/cim_operations.py

```diff
@@ -1172,12 +1237,302 @@ def __repr__(self):
                "default_namespace=%r, x509=%s, verify_callback=%r, " \
                "ca_certs=%r, no_verification=%r, timeout=%r, " \
                "use_pull_operations=%r, stats_enabled=%r, recorders=%s)" % \
-               (self.__class__.__name__, self.url, creds_repr, self.conn_id,
-                self.default_namespace, x509_repr, self.verify_callback,
-                self.ca_certs, self.no_verification, self.timeout,
-                self.use_pull_operations, self.stats_enabled,
+               (self.__class__.__name__, self.url, creds_repr,
+                self.conn_id, self.default_namespace, x509_repr,
+                self.verify_callback, self.ca_certs, self.no_verification,
+                self.timeout, self.use_pull_operations, self.stats_enabled,
                 recorder_list)
 
+    @classmethod
+    def configure_api_logger(cls, log_dest=DEFAULT_LOG_DESTINATION,
+                             detail_level=DEFAULT_API_LOG_DEFAULT_LEVEL,
+                             log_filename=DEFAULT_LOG_DESTINATION,
+                             connection=None):
+        """
+        A log configuration helper function to configure a logger  logs
+        execution/response/exceptions of the WBEMConnection methods that
+        communicate with a WBEM server.
+
+        Parameters:
+
+          log_dest (:term:`string`):
+            String defining the destination for this log. It must be one of the
+            destinations defined in :data:`~pywbem._logging.LOG_DESTINATIONS`
+            or `None`. If the value is the string ``none``, the null logger is
+            configured.
+
+          detail_level (:term:`string` or :class: `int`):
+            If :term:`string` it must be one of the key words that
+            defines the level of detail to be output. The possible
+            key words are 'all', 'summary', 'paths'.
+            TODO should get this list from variable.
+
+            If  :class: `int` it defines the maximum size of the log record
+            generated.
```

Comment:
The code also supports specifying the max size as a decimal string. This should be described.

Reply:
DONE removed